chalicelib/checks/wrangler_checks.py

```diff
@@ -1962,69 +1962,62 @@ def check_opf_lab_different_than_experiment(connection, **kwargs):
-                continue
-            exp_or_set = res['experiment_sets'][0]
+    other_processed_files = ff_utils.search_metadata(search, key=connection.ff_keys)
+
+    output_opfs = {'to_patch': [], 'problematic': []}
+    exp_set_uuids_to_check = []  # Exp or ExpSet uuid list
+    for opf in other_processed_files:
+        if opf.get('experiments'):
+            exp_or_sets = opf['experiments']
+        elif opf.get('experiment_sets'):
+            exp_or_sets = opf['experiment_sets']
         else:  # this should not happen
-            opf['problematic'].append({'@id': res['@id']})
+            output_opfs['problematic'].append({'@id': opf['@id']})
             continue
-        res['exp_set_uuid'] = exp_or_set['uuid']
-        if res['exp_set_uuid'] not in exp_set_uuids:
-            exp_set_uuids.append(res['exp_set_uuid'])
+        opf['exp_set_uuids'] = [exp_or_set['uuid'] for exp_or_set in exp_or_sets]
+        exp_set_uuids_to_check.extend([uuid for uuid in opf['exp_set_uuids'] if uuid not in exp_set_uuids_to_check])
 
     # get lab of Exp/ExpSet
-    result_exp_set = ff_utils.get_es_metadata(exp_set_uuids, sources=['uuid', 'properties.lab'], key=connection.ff_keys)
-    uuid_2_lab = {}  # map file uuid to Exp/Set lab
-    for item in result_exp_set:
-        uuid_2_lab[item['uuid']] = item['properties']['lab']
+    result_exp_set = ff_utils.get_es_metadata(exp_set_uuids_to_check, sources=['uuid', 'properties.lab'], key=connection.ff_keys)
+    es_uuid_2_lab = {}  # map Exp/Set uuid to Exp/Set lab
+    for es in result_exp_set:
+        es_uuid_2_lab[es['uuid']] = es['properties']['lab']
 
     # evaluate contributing lab
-    for res in result:
-        if res['@id'] not in [pr['@id'] for pr in opf['problematic']]:
-            contr_lab = []
-            exp_set_lab = uuid_2_lab[res['exp_set_uuid']]
-            if exp_set_lab == res['lab']['uuid']:
-                continue
-            elif res.get('contributing_labs'):
-                contr_lab = [lab['uuid'] for lab in res['contributing_labs']]
-                if exp_set_lab in contr_lab:
-                    continue
-            contr_lab.append(exp_set_lab)
-            opf['to_patch'].append({
-                '@id': res['@id'],
-                'contributing_labs': contr_lab,
-                'lab': res['lab']['display_title']})
-
-    if opf['to_patch'] or opf['problematic']:
+    for opf in other_processed_files:
+        if opf['@id'] in [opf_probl['@id'] for opf_probl in output_opfs['problematic']]:
+            # skip problematic files
+            continue
+        labs_to_add = []
+        opf_exp_set_labs = list(set([es_uuid_2_lab[exp_set] for exp_set in opf['exp_set_uuids']]))
+        if len(opf_exp_set_labs) == 1 and opf_exp_set_labs[0] == opf['lab']['uuid']:
+            # lab of exp/set is the same as lab of opf
+            continue
+        elif opf.get('contributing_labs'):
```

Comment:
Should there be an else clause?  I'm not fully sure on what is happening here (and I can pull down the branch to check if needed) but it seems that either you only have one lab and if it's the same then we're good or you have 'contributing_labs'.  Can there be cases when len(opf_exp_set_labs) !=1 or opf_exp_set_labs[0] != opf['lab'] and there are not 'contributing_labs' where you want to do something?

Reply:
You're right, this is a leftover from the previous code. It should be `else` instead.